akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/AskPattern.scala

```diff
@@ -117,6 +120,19 @@ object AskPattern {
             "native system is implemented: " + a.getClass)
       }
     }
+
+    /**
+     * The same as [[ask]] but only for requests that result in a response of type [[akka.actor.typed.StatusResponse]].
+     * If the status response is a [[StatusResponse.Fail]] the returned future will be failed, if it is a [[StatusResponse.Ok]]
+     * the returned future is completed successfully with the wrapped response.
+     */
+    def failableAsk[Res](
+        replyTo: ActorRef[StatusResponse[Res]] => Req)(implicit timeout: Timeout, scheduler: Scheduler): Future[Res] =
+      ask(replyTo).transform {
```

Comment:
if placed in `akka-actor` this can be in PromiseActorRef in same way as `Status`

Reply:
Sounds good, we'll need this signature anyway to get the return type right (the nested type)

Reply:
Actually, if we put it in classic ask won't it break typed ask, if you do that with the `StatusResponse` type, because it will magically unwrap and complete the future with the wrong type?

Reply:
so you mean using ordinary ask, and then sending back StatusResponse? isn't that a nice feature even? consistently failing the future for StatusResponse

we might have to amend the scaladoc, but it's already not completely accurate. it will fail the future also for a local actor that is terminated.

Reply:
I don't know... Someone else wrote an actor protocol and is forcing you to use a specific signature of ask but you won't know until runtime, when it fails. That doesn't sound nice to me.

Reply:
That's a good point.

Reply:
What I understood from this discussion, is that we won't have a `askWithStatus`, but only `ask` and document that it would automatically convert the status error to a failed future. 

Did I missed something?

Reply:
We can't do that and also unwrap the successful response message, because of the types, so it wouldn't get us all the way for typed, only for classic - and that would then break typed ask which uses classic ask internally.